daemon/state.go

```diff
@@ -96,9 +97,9 @@ func (d *Daemon) restoreOldEndpoints(dir string, clean bool) (*endpointRestoreSt
 		if ep.HasLabels(labels.LabelHealth) {
 			skipRestore = true
 		} else {
-			if ep.K8sPodName != "" && ep.K8sNamespace != "" {
+			if ep.K8sPodName != "" && ep.K8sNamespace != "" && k8s.Client() != nil {
```

Comment:
`k8s.IsEnabled()`

Reply:
Why is that better? It may still panic if the client is not initialized.

Reply:
@tgraf `k8s.IsEnabled()` is what we use on all code to check if cilium is running in k8s mode or not. `k8s.Client()` is never nil on both cases because the variable is already initialized.

Reply:
`k8s.IsEnabled()` is based on static configuration so how could it avoid a potential panic. It also doesn't mean that the apiserver is already reachable. I may have misunderstood your initial ask.

Reply:
I've changed it. It still doesn't work though. It's impossible to change fro k8s to non-k8s without cleaning all state.